app/src/main/java/com/kickstarter/viewmodels/CommentsViewHolderViewModel.kt

```diff
@@ -171,15 +181,32 @@ interface CommentsViewHolderViewModel {
                 .subscribe(this.flagComment)
         }
 
-        private fun isActionGroupVisible(commentCardData: CommentCardData, user: User?) =
-            commentCardData.project?.let {
-                it.isBacking || ProjectUtils.userIsCreator(it, user)
-            } ?: false
-
+        /**
+         * Checks if the current user is backing the current project
+         *  @param commentCardData
+         *  @param featureFlagActive
+         *
+         *  @return
+         *  true -> if current user is backer and the feature flag is active
+         *  false -> any of the previous conditions fails
+         */
+        private fun shouldReplyButtonBeVisible(
+            commentCardData: CommentCardData,
+            featureFlagActive: Boolean
+        ) =
+            commentCardData.project?.let { it.isBacking && featureFlagActive } ?: false
+
+        /**
+         * Updates the status of the current comment card.
+         * Also updates the current state of the [OptimizelyFeature.Key.COMMENT_ENABLE_THREADS]
+         * everytime the state changes.
+         */
         private fun cardStatus(comment: Comment?) = when {
             comment?.deleted() ?: false -> CommentCardStatus.DELETED_COMMENT
             (comment?.repliesCount() ?: false != 0) -> CommentCardStatus.COMMENT_WITH_REPLIES
             else -> CommentCardStatus.COMMENT_WITHOUT_REPLIES
+        }.also {
+            this.isCommentEnableThreads.onNext(optimizely.isFeatureEnabled(OptimizelyFeature.Key.COMMENT_ENABLE_THREADS))
```

Comment:
Please @hadia @sunday-okpoluaefe @leighdouglas note here the `also` block, as from this moment any update on the state of the comment card will also update the value for the feature flag